src/vmm/src/linux/vstate.rs

```diff
@@ -551,16 +561,31 @@ impl Vm {
         Ok(())
     }
 
+    pub fn guest_memfd_get(&self, gpa: u64) -> Option<(RawFd, u64)> {
+        self.guest_memfds.get(&gpa).copied()
+    }
+
+    #[allow(unused_mut)]
     fn memory_region_set(
         &mut self,
         guest_mem: &GuestMemoryMmap,
         region: &GuestRegionMmap,
     ) -> Result<()> {
         let host_addr = guest_mem.get_host_address(region.start_addr()).unwrap();
-        if !self.fd.check_extension(GuestMemfd) {
+        let start = region.start_addr().raw_value();
+        let end = start + region.len();
+
+        let mut use_guest_memfd = false;
+
+        #[cfg(feature = "tee")]
+        if self.fd.check_extension(GuestMemfd) {
+            use_guest_memfd = true;
+        }
+
+        if !use_guest_memfd {
```

Comment:
`check_extension()` already returns `bool`. Could we just do 
```rust
if !self.fd.check_extension(GuestMemfd) {
...
}
```

Reply:
With this, `use_guest_memfd` will always be false in non-TEE guests. With your suggestion, guest_memfd would be used even on non-TEE guests if supported. Do we want to do that? I'm not sure. For now, I'd say we shouldn't use guest_memfd when it's not required.

Reply:
The GuestMemfd capability is only enabled for SEV and KVM_SW_PROTECTED_VM types (for now). This should never be enabled on a non-TEE system. 

See: 
- https://github.com/torvalds/linux/blob/master/arch/x86/kvm/Kconfig
- https://github.com/torvalds/linux/blob/9d7a0577c9db35c4cc52db90bc415ea248446472/virt/kvm/kvm_main.c#L4845

Reply:
Good catch, then my `#[cfg(feature = "tee")]` is redundant.

Reply:
Fixed.